### DISCUSSION TITLE
Why do we need pages/api/auth/[...nextauth].ts in Nextjs app router?

### DISCUSSION THREAD

**Discussion by oasaleh** [nextauthjs/next-auth#9578] [Help]
I don't get why we need this route in the new Nextjs app router using Authjs v5. In Nextjs [learning docs](https://nextjs.org/learn/dashboard-app/adding-authentication), they don't even use it. Also, in Authjs docs, we are getting these handlers in `auth.ts` `handlers: { GET, POST }, auth` while in Nextjs learning docs, they just have these `auth, signIn, signOut`. What am I missing?

**Comment by mirkonasato** [comment]
You don't. Where did you see written that you need `pages/api` exactly? It's not on the page you linked.

**Reply by oasaleh** [reply]
Right here..

https://authjs.dev/guides/upgrade-to-v5
<img width="521" alt="image" src="https://github.com/nextauthjs/next-auth/assets/56449814/fdf3f77f-7b34-4432-8ac4-9d24ab415c40">


<img width="468" alt="Screenshot 2024-01-09 at 8 15 30 PM" src="https://github.com/nextauthjs/next-auth/assets/56449814/05eb1f65-02a4-4076-9168-cabe30a14849">

**Reply by mirkonasato** [reply]
That's `app/api`, not `pages/api` like you wrote in the title. But I understand what you mean now. I think the route handler is only needed if you use certain features, like the default sign in/out pages, and not if you call `signIn`/`signOut` in your own server actions like in the next-learn example.
